fifeforspark/base_modelers.py

```diff
@@ -0,0 +1,430 @@
+            max_lead_col (str): Name of the column representing the number of
+                observable future periods.
+            spell_col (str): Name of the column representing the number of
+                previous spells of consecutive observations of the same individual.
+            weight_col (str): Name of the column representing observation weights.
+            allow_gaps (bool): Whether or not observations should be included for
+                training and evaluation if there is a period without an observation
+                between the period of observations and the last period of the
+                given time horizon.
+        """
+        if (config.get("time_identifier", "") == "") and data is not None:
+            config["time_identifier"] = data.columns[1]
+
+        if (config.get("individual_identifier", "") == "") and data is not None:
+            config["individual_identifier"] = data.columns[0]
+
+        findspark.init()
+        self.spark = SparkSession.builder.getOrCreate()
+        self.config = config
+        self.data = data
+        self.duration_col = duration_col
+        self.event_col = event_col
+        self.predict_col = predict_col
+        self.test_col = test_col
+        self.validation_col = validation_col
+        self.period_col = period_col
+        self.max_lead_col = max_lead_col
+        self.spell_col = spell_col
+        self.weight_col = weight_col
+        self.allow_gaps = allow_gaps
+        self.reserved_cols = [
+            self.duration_col,
+            self.event_col,
+            self.predict_col,
+            self.test_col,
+            self.validation_col,
+            self.period_col,
+            self.max_lead_col,
+            self.spell_col,
+        ]
+        if self.config:
+            self.reserved_cols.append(self.config["individual_identifier"])
+        if self.weight_col:
+            self.reserved_cols.append(self.weight_col)
+        if self.data is not None:
+            self.categorical_features = [
+                col[0] for col in self.data.dtypes if col[1] == 'string']
+            self.numeric_features = [feature for feature in self.data.columns
+                                     if feature not in (self.categorical_features + self.reserved_cols)]
+            self.data = self.transform_features()
+
+    @abstractmethod
+    def train(self) -> Any:
+        """Train and return a model."""
+
+    @abstractmethod
+    def predict(
+        self, subset: Union[None, pyspark.sql.column.Column] = None, cumulative: bool = True
+    ) -> np.ndarray:
+        """Use trained model to produce observation survival probabilities."""
+
+    @abstractmethod
+    def evaluate(
+        self,
+        subset: Union[None, pyspark.sql.column.Column] = None,
+        threshold_positive: Union[None, str, float] = 0.5,
+        share_positive: Union[None, str, float] = None,
+    ) -> ks.frame.DataFrame:
```

Comment:
I had this output a spark dataframe in lgb modelers, want me to change that to a koalas dataframe?